src/charm.py

```diff
@@ -205,6 +197,44 @@ def _check_kf_profiles(self, interfaces):
 
         return kf_profiles
 
+    def main(self, event) -> None:
+        """Main entry point for the Charm."""
+        try:
+            self._check_container_connection()
+            self._check_model_name()
+            self._check_leader()
+            interfaces = self._get_interfaces()
```

Comment:
Feels more natural if we add `handle_ingress` here
```suggestion
            interfaces = self._get_interfaces()
            self.handle_ingress(interfaces)
```

Reply:
I put it there but I must disagree :) by convention the try execpt block should be arround the code where the exception is expected :) (if possible)